src/auxiliary/t8code.jl

```diff
@@ -0,0 +1,469 @@
+"""
+    init_t8code()
+
+Initialize `t8code` by calling `sc_init`, `p4est_init`, and `t8_init` while
+setting the log level to `SC_LP_ERROR`.  This function will check if `t8code`
+is already initialized and if yes, do nothing, thus it is safe to call it
+multiple times.
+"""
+function init_t8code()
+    t8code_package_id = t8_get_package_id()
+    if t8code_package_id >= 0
+        return nothing
+    end
+
+    # Initialize the sc library, has to happen before we initialize t8code.
+    T8code.Libt8.sc_init(mpi_comm(), 1, 1, C_NULL, T8code.Libt8.SC_LP_ERROR)
+
+    # Initialize `p4est` with log level ERROR to prevent a lot of output in AMR simulations
+    T8code.Libt8.p4est_init(C_NULL, SC_LP_ERROR)
```

Comment:
Is this the same `sc_init` and `p4est_init` that we get from P4est.jl? Will they interfere?

Reply:
No, these are Versions directly from `T8code.jl` resp. `t8code_jll`. That's why there is the full module path qualifier `T8code.Libt8` prepended in order to avoid name conflict with the versions imported by `P4est.jl`.